src/Microsoft.Azure.SignalR/EndpointProvider/DefaultServiceEndpointGenerator.cs

```diff
@@ -46,21 +50,27 @@ public string GetClientEndpoint(string hubName, string originalPath, string quer
                 queryBuilder.Append("&").Append(queryString);
             }
 
-            return $"{InternalGetEndpoint(ClientPath, hubName)}{queryBuilder}";
+            return $"{InternalGetEndpoint(ClientPath, hubName, ApplicationName)}{queryBuilder}";
         }
 
         public string GetServerAudience(string hubName) =>
-            InternalGetAudience(ServerPath, hubName);
+            InternalGetAudience(ServerPath, hubName, ApplicationName);
 
         public string GetServerEndpoint(string hubName) =>
-            InternalGetEndpoint(ServerPath, hubName);
+            InternalGetEndpoint(ServerPath, hubName, ApplicationName);
 
-        private string InternalGetEndpoint(string path, string hubName) =>
-            Port.HasValue ?
-            $"{Endpoint}:{Port}/{path}/?hub={hubName.ToLower()}" :
-            $"{Endpoint}/{path}/?hub={hubName.ToLower()}";
+        private string InternalGetEndpoint(string path, string hubName, string applicationName)
+        {
+            var prefixedHubName = string.IsNullOrEmpty(applicationName) ? hubName.ToLower() : $"{applicationName.ToLower()}_{hubName.ToLower()}";
```

Comment:
extract to a common method for calculate `prefixedHubName`?

Reply:
done